01PITT_INST-ULS_1_5/js/custom.js

```diff
@@ -267,45 +275,170 @@ angular.module('hathiTrustAvailability', []).constant('hathiTrustBaseUrl', 'http
+        $scope.validationMessage += 'The zip code is not formatted correctly.';
+      }
+      if ($scope.validationMessage != '') {
+        return;
+      }
+      $scope.statusMessage = 'Processing...';
+      $scope.currentAddress = null;
+      $scope.showInput = false;
+      let url = addressServiceBaseUrl + 'address/?jwt=' + self.getJwt();
+      $http.put(url, $scope.addressInput).then(() => {
+        $scope.getAddresses();
+        $scope.validationMessage = "";
+      }, error => {
+        $scope.showInput = true;
+        $scope.validationMessage = "An error occurred."
+      });
+    }
+
+    $scope.revertToHomeAddress = function() {
+      $scope.statusMessage = 'Processing...';
+      $scope.currentAddress = null;
+      let url = addressServiceBaseUrl + 'address/?jwt=' + self.getJwt();
+      $http.delete(url).then(() => {
+        $scope.getAddresses();
+      });
     }
-    //if they choose 'Ship it'...
-    if(typeof newVal == "string" && newVal.indexOf('$$USER_HOME_ADDRESS') > -1){
-      //display home delivery instructions:
-      angular.element(document.querySelector('#noContactRequestInstructions'))[0].style.display='none';
-      angular.element(document.querySelector('#shipItRequestInstructions'))[0].style.display='block';
-      //trying to set a 4-sided border for the comment field conflicts with the insufficient border set by ExL
-      //so highlight it in grey instead?..
-      angular.element(document.getElementsByTagName('prm-form-field'))[2].children[0].children[0].children[1].style.backgroundColor='lightgrey';
-      angular.element(document.getElementsByTagName('prm-form-field'))[2].children[0].children[0].children[1].style.opacity= "0.5";
+
+    $scope.showNextLine = function() {
+      $scope.shownAddressLines += 1;
     }
-    //if they choose one of our physical locations...
-    if (typeof newVal == "string" && newVal.indexOf('$$USER_HOME_ADDRESS') == -1){
-      //display no contact pickup instructions
-      angular.element(document.querySelector('#shipItRequestInstructions'))[0].style.display='none';
-      angular.element(document.querySelector('#noContactRequestInstructions'))[0].style.display='block';
-      //fix unecessary gap in instructions area:
-      //trying to set a 4-sided border for the comment field conflicts with the insufficient border set by ExL
-      //so highlight it in grey instead..
-      angular.element(document.getElementsByTagName('prm-form-field'))[2].children[0].children[0].children[1].style.backgroundColor='lightgrey';
-      angular.element(document.getElementsByTagName('prm-form-field'))[2].children[0].children[0].children[1].style.opacity= "0.5";
-      //$scope.shipit=false;
-    }       
+
+  }])
+  .component('addressSelector', {
+    require: {
+      primoExplore: '^primoExplore'
+    },
+    bindings: {parentCtrl: `<`},
+    controller: 'addressSelectorController',
+    template: 
+    '<div layout="row" ng-if="showAddresses()">\
```

Comment:
Yes! At first glance, this looks like a way cleaner strategy than the watcher-based one I employed.